src/components/business/hireTalent.tsx/skillDomainForm.tsx

```diff
@@ -0,0 +1,200 @@
+import React, { useState, useEffect } from 'react';
+import { PackageOpen } from 'lucide-react';
+import { useSelector } from 'react-redux'; // To get the user data
+
+import SkillDialog from './skillDiag';
+import DomainDialog from './domainDiag';
+
+import { Card } from '@/components/ui/card';
+import {
+  Table,
+  TableHeader,
+  TableBody,
+  TableRow,
+  TableHead,
+  TableCell,
+} from '@/components/ui/table';
+import { axiosInstance } from '@/lib/axiosinstance';
+import { Switch } from '@/components/ui/switch';
+import { RootState } from '@/lib/store';
+
+interface Skill {
+  _id: string;
+  label: string;
+}
+
+interface Domain {
+  _id: string;
+  label: string;
+}
+
+interface SkillDomainData {
+  uid: string;
+  label: string;
+  experience: string;
+  description: string;
+  status: string;
+  visible: boolean;
+}
+
+const SkillDomainForm: React.FC = () => {
+  const [skills, setSkills] = useState<Skill[]>([]);
+  const [domains, setDomains] = useState<Domain[]>([]);
+  const [skillDomainData, setSkillDomainData] = useState<SkillDomainData[]>([]);
+  const [statusVisibility, setStatusVisibility] = useState<boolean[]>([]);
+
+  // Get the user data from Redux store
+  const user = useSelector((state: RootState) => state.user);
+
+  // Fetch skills, domains, and user's skill/domain data on mount
+  useEffect(() => {
+    async function fetchData() {
+      try {
+        const skillsResponse = await axiosInstance.get('/skills/all');
```

Comment:
don't blindly believe you will get data successfully and data will be like response.data.data